packages/insomnia/src/ui/components/editors/environment-key-value-editor/key-value-editor.tsx

```diff
@@ -0,0 +1,344 @@
+              <Menu
+                aria-label='environment-kv-editor-type-selector'
+                selectionMode="single"
+                selectedKeys={[type]}
+                // Only valid json string or empty string allowed to convert to JSON type
+                disabledKeys={isValidJSONString || value.trim() === '' ? [] : [EnvironmentKvPairDataType.JSON]}
+                items={kvPairItemTypes}
+              >
+                {item => (
+                  <MenuItem
+                    key={item.id}
+                    id={item.id}
+                    className="aria-disabled:text-[--hl-lg] aria-disabled:cursor-not-allowed aria-disabled:bg-transparent flex gap-2 pl-[--padding-sm] pr-[--padding-xl] aria-selected:font-bold items-center text-[--color-font] h-[--line-height-xs] w-full
+                      text-md whitespace-nowrap bg-transparent hover:bg-[--hl-sm] focus:bg-[--hl-xs] focus:outline-none transition-colors text-sm react-aria-ListBoxItem"
+                    aria-label={item.name}
+                    onAction={() => handleItemChange(id, 'type', item.id)}
+                  >
+                    {({ isSelected }) => (
+                      <>
+                        <span>{item.name}</span>
+                        {isSelected && (
+                          <Icon
+                            icon="check"
+                            className="text-[--color-success] justify-self-end"
+                          />
+                        )}
+                      </>
+                    )}
+                  </MenuItem>
+                )}
+              </Menu>
+            </Popover>
+          </MenuTrigger>
+        </div>
+        <div className={`${cellCommonStyle} w-20`} >
+          <ItemButton
+            className="flex items-center justify-center h-7 aspect-square rounded-sm text-[--color-font] hover:bg-[--hl-xs] focus:ring-inset ring-1 ring-transparent focus:ring-[--hl-md] transition-all text-sm"
+            tabIndex={-1}
+            onPress={() => handleItemChange(id, 'enabled', !enabled)}
+          >
+            <Icon icon={enabled ? 'check-square' : 'square'} />
+          </ItemButton>
+          <PromptButton
+            disabled={kvPairs.length <= 1}
+            className="flex	items-center disabled:opacity-50 justify-center h-7 aspect-square aria-pressed:bg-[--hl-sm] rounded-sm text-[--color-font] hover:bg-[--hl-xs] focus:ring-inset ring-1 ring-transparent focus:ring-[--hl-md] transition-all text-sm"
+            fullWidth
+            confirmMessage=''
+            doneMessage=''
+            tabIndex={-1}
+            onClick={() => handleDeleteItem(id)}
+          >
+            <Icon icon="trash-can" />
+          </PromptButton>
+        </div>
+      </>
+    );
+  };
+
+  useEffect(() => {
+
+  }, []);
+
+  return (
+    <div className="p-[--padding-sm] min-w-max h-full overflow-hidden flex flex-col">
```

Comment:
Looks like there is extra padding around the toolbar. If we want we could add padding in the key-value editor ListBox bellow
```suggestion
    <div className="min-w-max h-full overflow-hidden flex flex-col">
```

Reply:
OK, will remove the padding for toolbar